### ISSUE TITLE
Quaternion slerp returning nan values

### ISSUE THREAD

**Issue by SlidyBat** [redorav/hlslpp#68]
Found that slerp can sometimes return nan values, caused by a division by 0 in the implementation.
To replicate:
```cpp
auto r = slerp(
	quaternion( -0.0112188980, -0.0367633253, -0.00361495349, -0.999254525 ),
	quaternion( -0.0114078531, -0.0367971063, -0.00342923636, -0.999251783 ),
	0.691265166
);
assert( !any( isnan( r ) ) );
```

**Comment by redorav** [comment]
I'll take a look asap today. Thanks for reporting and a good repro case

**Comment by redorav** [comment]
Hi @SlidyBat I've fixed this in b19c81b88600a500900ac637fc43ad1409494778

There were two issues, one with vectors where the dot product is slightly beyond the range of acos, the other when the angle is 0 and we divide by it, both of which have been fixed and don't add that much cost.

After reading further about slerp I realized that most people seem to use a normalized lerp they call nlerp which has different properties but is a lot faster. If this is useful to you and you want to give it a go, there's an nlerp function now. I've also kept lerp in case you want to lerp multiple times, then normalize at the end which obviously is faster than calling nlerp multiple times but is faster.

Hopefully this fixes your issue.
